test/tstRay.cpp

```diff
@@ -225,3 +225,32 @@ BOOST_AUTO_TEST_CASE(overlap_distance_sphere,
   BOOST_TEST(overlapDistance(Ray{{half_sqrtf_3, 0.5, 0}, {1, 0, 0}},
                              unit_sphere) == 0.f);
 }
+
+#define STATIC_ASSERT(cond) static_assert(cond, "");
+BOOST_AUTO_TEST_CASE(make_euclidean_vector)
+{
+  using ArborX::Experimental::makeVector;
+  using ArborX::Experimental::Vector;
+  STATIC_ASSERT((makeVector({0, 0, 0}, {1, 2, 3}) == Vector{1, 2, 3};));
+  STATIC_ASSERT((makeVector({1, 2, 3}, {4, 5, 6}) == Vector{3, 3, 3};));
```

Comment:
Why do you use `static_assert` instead of `BOOST_TEST` in all newly introduced tests?

Reply:
Because it can be checked at compile time